src/db/parity_db.rs

```diff
@@ -3,15 +3,69 @@
 
 use std::{path::PathBuf, sync::Arc};
 
+use super::SettingsStore;
+
+use crate::db::{parity_db_config::ParityDbConfig, DBStatistics};
 use crate::libp2p_bitswap::{BitswapStoreRead, BitswapStoreReadWrite};
-use anyhow::anyhow;
+
+use anyhow::{anyhow, Context};
+use cid::multihash::Code::Blake2b256;
+
 use cid::Cid;
+
 use fvm_ipld_blockstore::Blockstore;
+
+use fvm_ipld_encoding::DAG_CBOR;
+
 use parity_db::{CompressionType, Db, Operation, Options};
+use strum::{Display, EnumIter, FromRepr, IntoEnumIterator};
+
 use tracing::warn;
 
-use super::errors::Error;
-use crate::db::{parity_db_config::ParityDbConfig, DBStatistics, Store};
+/// This is specific to Forest's `ParityDb` usage.
+/// It is used to determine which column to use for a given entry type.
+#[derive(Copy, Clone, Debug, Display, PartialEq, FromRepr, EnumIter)]
+#[repr(u8)]
+enum DbColumn {
+    /// Column for storing IPLD data with `Blake2b256` hash and `DAG_CBOR` codec.
+    /// Most entries in the `blockstore` will be stored in this column.
+    GraphDagCborBlake2b256,
+    /// Column for storing other IPLD data (different codec or hash function).
+    /// It allows for key retrieval at the cost of degraded performance. Given that
+    /// there will be a small number of entries in this column, the performance
+    /// degradation is negligible.
+    GraphFull,
+    /// Column for storing Forest-specific settings.
+    Settings,
+}
+
+impl DbColumn {
+    fn create_column_options(compression: CompressionType) -> Vec<parity_db::ColumnOptions> {
+        DbColumn::iter()
+            .map(|col| {
+                match col {
+                    DbColumn::GraphDagCborBlake2b256 => parity_db::ColumnOptions {
```

Comment:
` DbColumn::GraphDagCborBlake2b256 | DbColumn::Settings`

Reply:
I like it. Thanks.